base/markdown/GitHub/GitHub.jl

```diff
@@ -33,7 +33,83 @@ function github_paragraph(stream::IO, md::MD, config::Config)
     return true
 end
 
-# TODO: tables
+typealias Row Vector
 
-@flavor github [list, indentcode, blockquote, fencedcode, hashheader, github_paragraph,
-                 en_dash, inline_code, asterisk_bold, asterisk_italic, image, link]
+type Table
+    rows::Vector{Row}
+    align
+end
+
+function github_table(stream::IO, md::MD, config::Config)
+    withstream(stream) do
+        rows = Row[]
+        n = 0
+        align = :r # default is to align right
+        while !eof(stream)
+            n += 1
+            pos = position(stream)
```

Comment:
Since you're repeating the `withstream` pattern here by hand, I'm thinking it might be best to factor row parsing into its own `parse_row` function, or something similar. e.g.
```julia
function parse_row(stream::IO)
  withstream(stream) do
    # return row vector or `nothing`, without explicitly seeking the stream
```
Then just call `parse_row` until it returns `nothing`.